torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1077,7 +1076,9 @@ class PyLoweringContext {
       at::ScalarType dtype =
           MaybeUpcastToHostTorchType(literal.shape().element_type());
       at::Tensor input = MakeTensorFromXlaLiteral(literal, dtype);
-      results[param_ids[i]] = input;
+      std::optional param_id = lowering_ctx.GetParameterId(device_data[i]);
+      XLA_CHECK(param_id.has_value());
```

Comment:
when would it not has value?

Reply:
When `GetParameterId` receives a `BackendData` that is not a parameter in this lowering context, it will return `std::nullopt`. However, this loop is only iterating over parameters (line 1071, `const std::vector<torch::lazy::BackendDataPtr>& device_data = lowering_ctx.GetParametersData();`), so we will expect all `BackendData` there to have an ID. Seems good to enforce this invariant.